docs/how-tos/hello-world-module.md

```diff
@@ -0,0 +1,508 @@
+
+    ```python {class="line-numbers linkable-line-numbers" data-start="94" }
+    MODEL: ClassVar[Model] = Model(ModelFamily("jessamy", "hello-world"), "hello-sensor")
+    ```
+
+1. Add the imports that are unique to the sensor file:
+
+   ```python {class="line-numbers linkable-line-numbers"}
+   from viam.components.sensor import *
+   from viam.utils import SensorReading
+   ```
+
+   Save the <file>hello-world/src/main.py</file> file.
+
+1. Open <file>temporary/meta.json</file> and copy the model information.
+   For example:
+
+   ```json {class="line-numbers linkable-line-numbers" data-start="8"}
+    {
+      "api": "rdk:component:sensor",
+      "model": "jessamy:temporary:hello-sensor"
+    }
+   ```
+
+1. Open <file>hello-world/meta.json</file> and paste the sensor model into the model list.
+
+   Edit the `description` to accurately include both models.
+
+   Change `temporary` to `hello-world`.
+
+   The file should now resemble the following:
+
+    ```json {class="line-numbers linkable-line-numbers" data-line="6-16"}
+    {
+      "$schema": "https://dl.viam.dev/module.schema.json",
+      "module_id": "jessamy:hello-world",
+      "visibility": "private",
+      "url": "",
+      "description": "Example camera and sensor components: hello-camera and hello-sensor",
+      "models": [
+        {
+          "api": "rdk:component:camera",
+          "model": "jessamy:hello-world:hello-camera"
+        },
+        {
+          "api": "rdk:component:sensor",
+          "model": "jessamy:hello-world:hello-sensor"
+        }
+      ],
+      "entrypoint": "./run.sh",
+      "first_run": ""
+    }
+    ```
+
+1. You can now delete the <file>temporary</file> module directory and all its contents.
+
+{{< /expand >}}
+
+## Implement the API methods
+
+Edit the stub files to implement your test script in a way that works with the camera and sensor APIs:
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+### Implement the camera API
+
+First, implement the camera API methods by editing the camera class definition:
+
+1. Open the <file>hello-world/src/main.py</file> file again in your code editor.
+
+1. Add the following to the list of imports at the top of <file>main.py</file>:
+
+    ```python {class="line-numbers linkable-line-numbers"}
+    from viam.media.utils.pil import pil_to_viam_image
+    from viam.media.video import CameraMimeType
+    from viam.utils import struct_to_dict
+    from PIL import Image
+    ```
+
+1. In the test script you hard-coded the path to the image.
+   For the module, let's make the path a configurable attribute so you or other users of the module can set a different path.
+   Add the following lines to the camera's `reconfigure()` function definition:
+
+    ```python {class="line-numbers" data-start="68"}
+    attrs = struct_to_dict(config.attributes)
+    self.image_path = str(attrs.get("image_path"))
+    ```
+
+1. Since the camera needs to get an image from somewhere, `image_path` is a required attribute.
```

Comment:
One doesn't really logically follow from the other. The camera could get the image from a default path or w/e. It's a choice to do it like this.
```suggestion
1. We are not providing a default image but rely on the end user to supply a valid path to an image when configuring the resource.
   This means `image_path` is a required attribute.
```